engine/consensus/compliance/engine.go

```diff
@@ -19,98 +17,126 @@ import (
 	"github.com/onflow/flow-go/module/metrics"
 	"github.com/onflow/flow-go/module/trace"
 	"github.com/onflow/flow-go/network"
-	"github.com/onflow/flow-go/state"
 	"github.com/onflow/flow-go/state/protocol"
 	"github.com/onflow/flow-go/storage"
+	"github.com/onflow/flow-go/utils/fifoqueue"
 	"github.com/onflow/flow-go/utils/logging"
 )
 
-// Engine is the consensus engine, responsible for handling communication for
-// the embedded consensus algorithm.
+type Event struct {
+	OriginID flow.Identifier
+	Msg      interface{}
+}
+
+type (
+	EventSink chan *Event // Channel to push pending events
+)
+
+// defaultBlockQueueCapacity maximum capacity of block proposals queue
+const defaultBlockQueueCapacity = 10000
+
+// defaultVoteQueueCapacity maximum capacity of block votes queue
+const defaultVoteQueueCapacity = 10000
```

Comment:
I would substantially reduce this Storing more than 1 round of votes is generally not useful, because as soon as a view is over, the votes are largely useless. 

We currently have less than 150 consensus nodes. Setting the queue length to maybe 1000 would give ample room for a couple views and many duplicates.